### PR TITLE
Assign random available priority to listener rule

### PR DESCRIPTION
**Pull request by Equlnox**
JIRA: https://rippling.atlassian.net/browse/INFRA-1049
Container runner PR: https://github.com/Rippling/container-runner-api/pull/7

### CODE REVIEW

File: cloudlift/config/service_configuration.py [hunk 1]
```diff
@@ -101,13 +104,50 @@ def get_config(self, strip_cloudlift_version=True):
...
 
-    def set_config(self, config):
+    def set_config(self, config, information_fetcher=None):
         '''
             Set configuration in DynamoDB
         '''
         config['cloudlift_version'] = VERSION
+        for service_name, service_config in config['services']:
+            reuse_existing_alb = 'alb' in service_config['http_interface'] and service_config['http_interface']['alb'].get('create_new', False)
+            if reuse_existing_alb and 'priority' not in config['http_interface']['alb']['priority']:
```
Comment (aswinkarthik) [inline cloudlift/config/service_configuration.py:114]: This seems wrong. 
```
'priority' not in config['http_interface']['alb']['priority']:
```
Should it be

```
'priority' not in config['http_interface']['alb']:
```

can you add a unit tests for this?

File: cloudlift/config/service_configuration.py [hunk 2]
```diff
@@ -101,13 +104,50 @@ def get_config(self, strip_cloudlift_version=True):
...
+                if matching_priority:
+                    return matching_priority
+
+        return self._get_random_available_listener_rule_priority(listener_rules)
+
+    def _get_random_available_listener_rule_priority(self, listener_rules):
+        occupied_priorities = set(rule['Priority'] for rule in listener_rules)
+        available_priorities = set(range(1, 50000)) - occupied_priorities
+        if not available_priorities:
+            return -1
```
Comment (aswinkarthik) [inline cloudlift/config/service_configuration.py:148]: shall we raise Exception here instead?

File: cloudlift/deployment/service_information_fetcher.py [hunk 3]
```diff
@@ -124,6 +127,21 @@ def _fetch_current_image_uri(self):
...
+        while 'NextToken' in response:
+            response = self.cfn_client.list_stack_resources(
+                StackName=stack_name,
+                NextToken=response['NextToken'],
+            )
+            resource_summaries.extend(response.get('Rules', []))
+        return resource_summaries
+
+    def get_existing_listener_rule_summary(self, service_name):
+        return next((rule for rule in self.listener_rules if rule['LogicalResourceId'].beginswith(service_name)), None)
```
Comment (aswinkarthik) [inline cloudlift/deployment/service_information_fetcher.py:143]: Would `next` error out if self.listener_rules is empty list?

**Review by aswinkarthik — CHANGES_REQUESTED** [review]

File: cloudlift/deployment/service_information_fetcher.py (hunk 3 above)
Comment (Equlnox) [reply cloudlift/deployment/service_information_fetcher.py:143]: It would return `None`

File: cloudlift/config/service_configuration.py (hunk 2 above)
Comment (Equlnox) [reply cloudlift/config/service_configuration.py]: Raising exception now

**Review by aswinkarthik — APPROVED** [review]

**Pull request merged**